src/main/java/ipman/models/Task.java

```diff
@@ -0,0 +1,46 @@
+package ipman.models;
+
+/**
+ * Represents a task that has a name and may or may not be completed.
+ */
+public abstract class Task {
+    protected boolean isDone = false;
```

Comment:
just curious, how did you decide on these access modifiers?

Reply:
In this case, I consider these fields to belong mainly to the classes that inherit this Task class (since Task can't be instantiated anyways). I think that it should be "private" to the subclass i.e. protected.

On the other hand, I could very well just expose `isDone` with a getter and disallow subclasses from modifying it. I'll update this to private and implement getters so that testing is easier. I will be able to check if a Task is marked by some command if I had a getter anyways.

Reply:
Resolved by https://github.com/nus-cs2103-AY2425S1/ip/pull/538/commits/abf860ce968af86f71947ba360d1db9783c63039